frontend/src/containers/OnBoarding.js

```diff
@@ -55,12 +54,17 @@ export class OnBoarding extends Component {
       githubUsername,
       fetchReposFromGitHub,
       appendGithubForm,
-      params
+      params,
+      notify
     } = this.props;
 
-    if (githubUsername && this.state.step === 2) {
+    if (githubUsername && this.state.step === 1) {
       appendGithubForm({username: githubUsername, token: params.token});
-      fetchReposFromGitHub(githubUsername);
+      fetchReposFromGitHub(githubUsername)
+      .catch((error) => {
+        this.setState({step: 0});
```

Comment:
Won't this create an infinite loop? When user goes back to first page, clicking on Github instantly auths them and they are back to the first screen with same notification. I think it'll be better to give them that message inside the flow. What do you think?